golf2/Build.mk

```diff
@@ -0,0 +1,40 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+.PHONY: golf2/build
+golf2/build: golf2/target/thumbv7m-none-eabi/release/golf2
```

Comment:
On one hand, the fact that the directory structure is explicitly bound in the rules is brittle. On the other, seeing how  the default Tock make system does all kinds of macro stuff that then makes it frustrating to debug, I agree with simpler, explicit approaches. :)

Reply:
> On one hand, the fact that the directory structure is explicitly bound in the rules is brittle.

This is exactly how `Bazel` works, and it isn't a problem in practice. Granted, we have some amount of internal automation that helps move things, but even writing everything by hand remains manageable.

Conceptually, the *purpose* of the $DIR/{build,check,run,...} rules is "build everything under $DIR" so the name is actually fairly descriptive of what it does, and would be invalidated if the file was moved regardless.

But yes, if you move code around you need to update the build system.

Reply:
Sounds good to me!